app/TuistApp/Sources/Views/SimulatorsView/SimulatorsView.swift

```diff
@@ -0,0 +1,88 @@
+import Path
+import SwiftUI
+import TuistCore
+import TuistServer
+import TuistSupport
+
+struct SimulatorsView: View, ErrorViewHandling {
+    @State var viewModel = SimulatorsViewModel()
+    @EnvironmentObject var errorHandling: ErrorHandling
+    @State var isExpanded = false
+
+    init(appDelegate: AppDelegate) {
+        // We can't rely on the SimulatorsView to be rendered before a deeplink is triggered.
+        // Instead, we listen to the deeplink URL through an `AppDelegate` callback
+        // that's eagerly set up in this `init` on startup.
+        let viewModel = viewModel
+        Task {
+            do {
+                try await viewModel.onAppear()
+            } catch {
+                ErrorHandling().handle(error: error)
+            }
+        }
+        appDelegate.onChangeOfURL = { url in
+            Task {
+                do {
+                    try await viewModel.onChangeOfURL(url)
+                } catch {
+                    ErrorHandling().handle(error: error)
```

Comment:
What about having a `let errorHandling = ErrorHandling()` instance that you can reuse throughout this `init`?